source/solvers/vof.cc

```diff
@@ -647,11 +646,14 @@ VolumeOfFluid<dim>::find_sharpening_threshold()
     this->simulation_parameters.multiphysics.vof_parameters.conservation
       .monitored_fluid;
 
-  double mass_deviation = 0.;
-  int    nb_search_ite  = 0;
-  double st_ave         = 0.;
+  int nb_search_ite = 0;
   // Local variable for the tested sharpening_threshold values
-  double st_tested = 0.;
+
+  double st_avg             = (st_min + st_max) / 2.;
+  double mass_deviation_avg = calculate_mass_deviation(monitored_fluid, st_avg);
```

Comment:
if it's only for the output, better to put it after `mass_deviation_avg` is computed in the do-loop rather than compute it two times for the first iteration

Reply:
Done